tools/metadata_tools/sample_metadata.py

```diff
@@ -0,0 +1,581 @@
+        replacements["$$embedded_resources$$"] = "" # TODO
+        replacements["$$nuget_packages$$"] = get_csproj_xml_for_nuget_packages(self.nuget_packages)
+        replacements["$$code_and_xaml$$"] = get_csproj_xml_for_code_files(all_source_files, platform)
+        replacements["$$axml_files$$"] = get_csproj_xml_for_android_layout(all_source_files)
+        replacements["$$current_year$$"] = str(datetime.now().year)
+        replacements["$$friendly_name$$"] = self.friendly_name
+
+        # rewrite files in output - replace template fields
+        sample_metadata.rewrite_files_in_place(output_dir, replacements)
+
+        # write out the sample file
+        self.emit_dot_sample_file(platform, output_dir)
+
+        return
+    
+    def emit_dot_sample_file(self, platform, output_dir):
+        output_xml = "<ArcGISRuntimeSDKdotNetSample>\n"
+        # basic metadata
+        output_xml += f"\t<SampleName>{self.formal_name}</SampleName>\n"
+        output_xml += f"\t<SampleDescription>{self.description}</SampleDescription>\n"
+        output_xml += f"\t<ScreenShot>{self.images[0]}</ScreenShot>\n"
+        # code files, including XAML
+        output_xml += "\t<CodeFiles>\n"
+        for source_file in self.source_files:
+            output_xml += f"\t\t<CodeFile>{source_file}</CodeFile>\n"
+        output_xml += "\t</CodeFiles>\n"
+        
+        # xaml files
+        output_xml += "\t<XAMLParseFiles>\n"
+        for source_file in self.source_files:
+            if source_file.endswith(".xaml"):
+                output_xml += f"\t\t<XAMLParseFile>{source_file}</XAMLParseFile>\n"
+        output_xml += "\t</XAMLParseFiles>\n"
+
+        # exe
+        if platform == "WPF":
+            output_xml += "\t<DllExeFile>bin\debug\ArcGISRuntime.exe</DllExeFile>\n"
+        elif platform == "UWP" or platform == "XFU":
+            output_xml += "\t<DllExeFile>obj\\x86\Debug\intermediatexaml\ArcGISRuntime.exe</DllExeFile>\n"
+        elif platform == "Android" or platform == "XFA":
+            output_xml += "\t<DllExeFile>bin\debug\ArcGISRuntime.dll</DllExeFile>\n"
+        elif platform == "iOS" or platform == "XFI":
+            output_xml += "\t<DllExeFile>bin\iPhone\debug\ArcGISRuntime.exe</DllExeFile>\n"
+        
+        output_xml += "</ArcGISRuntimeSDKdotNetSample>\n"
+
+        filename = os.path.join(output_dir, f"{self.formal_name}.sample")
+
+        safe_write_contents(filename, output_xml)
+    
+    def populate_snippets_from_folder(self, platform, path_to_readme):
+        '''
+        Take a path to a readme file
+        Populate the snippets from: any .xaml, .cs files in the directory; 
+        any .axml files referenced from .cs files on android
+        '''
+        # populate files in the directory
+        sample_dir = os.path.split(path_to_readme)[0]
+        for file in os.listdir(sample_dir):
+            if os.path.splitext(file)[1] in [".axml", ".xaml", ".cs"]:
```

Comment:
This precludes snippets from additional code files used by the sample (i.e. location tracker sample). Might be relevant in future samples/snippets.